problem 1 :
Greg was 14, Marcia was 12, Peter was 11, Jan was 10, Bobby was 8, and Cindy was 6 when they started playing the Brady kids on The Brady Bunch. Cousin Oliver was 8 years old when he joined the show. What are the mean, median, and mode of the kids' ages when they first appeared on the show? What are the variance, standard deviation, and standard error?

In [67]:
import pandas as pd
import numpy as np

kids = pd.DataFrame(index = ["Greg", "Marcia", "Peter", "Jan", "Bobby", "Cindy", "Oliver"])
kids["age"] = [14, 12, 11, 10, 8, 6, 8]

kids.age.mode()

def summary_of_sorts(df):
    print("mean is {}".format(df.mean()))
    print("median is {}".format(df.median()))
    print("mode is \n{}".format(df.mode()))

    #why is there a zero???

    print("variance: {}".format(df.var()))
    print("standard deviation: {}".format(np.std(df,ddof=1)))

    print("standard error: {}".format(df.var()/(len(df)-1) ))
    pass
    
summary_of_sorts(kids)


mean is age    9.857143
dtype: float64
median is age    10.0
dtype: float64
mode is 
   age
0    8
variance: age    7.47619
dtype: float64
standard deviation: age    2.734262
dtype: float64
standard error: age    1.246032
dtype: float64


problem 2:
    Using these estimates, if you had to choose only one estimate of central tendency and one estimate of variance to describe the data, which would you pick and why?

In [ ]:
either mean or median would be ok given there are no outliers. 

problem 3:
Next, Cindy has a birthday. Update your estimates- what changed, and what didn't?

In [42]:
kids.loc["Cindy"] = 7
kids.loc["Cindy"]

summary_of_sorts(kids)

#*mean increased, variance lowered, std is also lower, standard error is also lower
#median and mode remained the same

mean age is 10.0
median age is 10.0
mode age is 0    8
dtype: int64
variance of age is 6.333333333333333
standard deviation of age is 2.516611478423583
standard error of age is 1.0555555555555556


problem4:
    Nobody likes Cousin Oliver. Maybe the network should have used an even younger actor. Replace Cousin Oliver with 1-year-old Jessica, then recalculate again. Does this change your choice of central tendency or variance estimation methods?

In [57]:
names = kids.index.tolist()
oliver_index = names.index("Oliver")
names[oliver_index] = "Jessica"
kids.index = names
kids.loc["Jessica"] = 1
kids


age
Greg      14
Marcia    12
Peter     11
Jan       10
Bobby      8
Cindy      6
Jessica    1

In [58]:
summary_of_sorts(kids)
#jessica is now an outlier, I'd prefer the median over the mean now

mean age is 8.857142857142858
median age is 10.0
mode age is 
0     1
1     6
2     8
3    10
4    11
5    12
6    14
dtype: int64
variance of age is 18.80952380952381
standard deviation of age is 4.3369947901195145
standard error of age is 3.134920634920635


problem5:
    On the 50th anniversary of The Brady Bunch, four different magazines asked their readers whether they were fans of the show. The answers were: TV Guide 20% fans Entertainment Weekly 23% fans Pop Culture Today 17% fans SciPhi Phanatic 5% fans

Based on these numbers, what percentage of adult Americans would you estimate were Brady Bunch fans on the 50th anniversary of the show?

In [68]:
fans = pd.DataFrame(index = ["TV Guide", "Entertainment Weekly", "Pop Culture", "SciPhi Phanatic"])
fans["percent as fans"] = [.2, .23, .17, .05]
fans
summary_of_sorts(fans)

mean is percent as fans    0.1625
dtype: float64
median is percent as fans    0.185
dtype: float64
mode is 
   percent as fans
0             0.05
1             0.17
2             0.20
3             0.23
variance: percent as fans    0.006225
dtype: float64
standard deviation: percent as fans    0.078899
dtype: float64
standard error: percent as fans    0.002075
dtype: float64


if fans of four entertainment magazines can be used as a reliable estimate for the u.s population, a reasonal guesstimate would be 18.5% of the population are fans of the show